src/ot_orchestration/dags/config/unified_pipeline.py

```diff
@@ -105,26 +124,62 @@ def init_pis_config(self) -> dict[str, Any]:
 
         return pis_raw_conf
 
+    def pis_env_vars(self, step_name: str) -> dict[str, str]:
+        """Return the environment variables for a PIS step."""
+        return {
+            "PIS_STEP": step_name.replace("pis_", ""),
+            "PIS_CONFIG_FILE": "/config.yaml",
+            "PIS_POOL": str(self.pis_pool),
+        }
+
+    def ontoform_args(self, step_name: str) -> list[str]:
+        """Return the arguments for the ONTOFORM step."""
+        real_step_name = step_name.replace("ontoform_", "")
+        return ["--work-dir", self.gcs_url, real_step_name]
+
     # pyhocon returns a ConfigTree, but we can treat it as a dict
-    def init_etl_config(self) -> Any:
+    def init_etl_config(self) -> dict[str, Any]:
         """Initialize the ETL configuration.
 
         This method reads the ETL configuration file, replaces the fields defined
-        in the orchestrator config, and returns the resulting configuration.
+        in the unified pipeline config, and returns the resulting configuration.
         """
-        etl_raw_conf = read_hocon_config(self.etl_config_local_path)
-
-        # set the work bucket paths
-        etl_raw_conf["common"]["input"] = f"{self.gcs_url}/input"
-        etl_raw_conf["common"]["output-base_path"] = f"{self.gcs_url}/output"
-
-        # fill in the input data versions
-        etl_raw_conf["chembl_version"] = self.chembl_version
-        etl_raw_conf["ensembl_version"] = self.ensembl_version
+        etl_raw_conf = read_hocon_config(
+            self.etl_config_local_path,
+            sentinels={
+                "gcs_url": self.gcs_url,
+                "release": self.release,
+                "chembl_version": self.chembl_version,
+                "ensembl_version": self.ensembl_version,
+            },
+        )
 
         # ppp - set the write mode to overwrite and remove the data sources
         if self.is_ppp:
             etl_raw_conf["spark-settings"]["write-mode"] = "overwrite"
             etl_raw_conf["evidences"]["data-sources-exclude"] = []
 
         return etl_raw_conf
+
+    def init_gentropy_settings(self) -> dict[str, Any]:
+        """Initialize the gentropy configuration.
+
+        This method reads the gentropy configuration file, replaces the fields defined
+        in the unified pipeline config, and returns the resulting configuration.
+        """
+        return read_yaml_config(
+            self.gentropy_config_local_path,
+            sentinels={
+                "gcs_url": self.gcs_url,
+                "gentropy_version": self.gentropy_version,
+                "vep_version": self.vep_version,
+            },
+        )
+
+    def gentropy_step(self, step_name: str) -> dict[str, Any]:
```

Comment:
Anything specific you thing does not belong into configuration initialization?

Reply:
It is a general rule when writing python classes
Ideally init should just set up the object properties, do not perform any computations. 
* This makes testing easier if the `read_config` and any parsers are collected into common functions.
* It aligns with OOP to move the computations to methods and perform the side effects there.
* The code is very imperative in this way, harder to follow.

Reply:
I'm not 100% with you there. Can you show me where it is stated that only assigning properties in a constructor is good practice in Python?

The current design ensures the object is fully initialized and valid after construction, which is a good practice.

I think there are no side effects in the function, if there are any please point them to me. If you think about it, all it does is:

* Read config files it needs to initialize the object
* Derive values from those (like image URLs, etc)
* Compute list of steps

Regarding the code being very imperative, remember python's principle: _explicit is better than implicit_.

Reply:
>  `explicit is better than implicit` 
implict is not the same as imperative. You can still be explicit about the steps used, while contain the business logic in  declarative methods.
* read config files is a side effect, will break if the config is not there (especialy that we raise error in the read function), this will prevent you from initializing the object.
> ensuring the object is fully validated is a good practice I agree, but in this case the constructor do most of the heavy lifting of the config processing, the class can not be easily extended by addition of the new child configuration without code modification... (in summary violates  SOLID principles)

The code is logically correct and works (from your experience), we can leave it this way if you do not feel convinced to change it.

Reply:
* If the config is not there, this must break. It should break when it initializes.
* I agree reading files is a side effect, but int he strictest "pure" function meaning. The init is not changing any other parts of the application state though.
* It cannot be extended easily by adding new child configs. But I do not expect that to happen often, so I did not want to model a modular config loader class. That requires a lot more work and complexity.